### PR TITLE
Don't kill substring matches

### PR DESCRIPTION
**Pull request by manics**
A substring match is defined as a string with a alphanumeric or underscore character before or after the match.

Alternatively we could have two configured lists, one for substring matches (mostly useful for short strings as these are more likely to match with valid strings) and that behaves as at present.

### CODE REVIEW

File: scripts/execwhacker.py [hunk 1]
```diff
@@ -67,6 +67,14 @@ def kill_if_needed(banned_strings_automaton, allowed_patterns, cmdline, pid, sou
                             f"action:spared pid:{pid} cmdline:{cmdline} matched:{b} allowed-by:{ap} source:{source.value}"
                         )
                     return
+            if (
+                    re.search(r"\w" + re.escape(b), cmdline, re.IGNORECASE) or
```
Comment (yuvipanda) [inline scripts/execwhacker.py:71]: Can you add a comment here explaining why we do this?

File: scripts/execwhacker.py [hunk 2]
```diff
@@ -67,6 +67,14 @@ def kill_if_needed(banned_strings_automaton, allowed_patterns, cmdline, pid, sou
...
+                    re.search(re.escape(b) + r"\w", cmdline, re.IGNORECASE)
+            ):
+                logging.info(
+                    f"action:spared pid:{pid} cmdline:{cmdline} matched:{b} allowed-by:<is-substring> source:{source.value}"
```
Comment (yuvipanda) [inline scripts/execwhacker.py:75]:
```suggestion
                    f"action:spared pid:{pid} cmdline:{cmdline} matched:{b} allowed-by:is-substring source:{source.value}"
```

**Review by yuvipanda — APPROVED** [review]
Couple of suggestions, but let's try it to see. May help reduce amount of tuning necessary!

File: scripts/execwhacker.py (hunk 1 above)
Comment (manics) [reply scripts/execwhacker.py:73]: Added

**Comment by yuvipanda** [comment]
Thanks @manics

**Pull request merged**